crates/solidity/inputs/language/src/definition.rs

```diff
@@ -6679,448 +6022,870 @@ codegen_language_macros::compile!(Language(
+                ),
+                BuiltInType(
+                    name = "%StringType",
+                    fields = [],
+                    functions = [BuiltInFunction(
+                        name = "concat",
+                        parameters = ["string[] stringsToConcatenate"],
+                        return_type = "string memory"
+                    )]
+                ),
+                BuiltInType(
+                    name = "%TransactionType",
+                    fields = [
+                        BuiltInField(definition = "uint gasprice"),
+                        BuiltInField(
+                            definition = "address payable origin",
+                            enabled = Till("0.8.0")
+                        ),
+                        BuiltInField(definition = "address origin", enabled = From("0.8.0"))
+                    ],
+                    functions = []
+                ),
+                BuiltInType(
+                    name = "%ContractTypeType",
+                    fields = [
+                        BuiltInField(definition = "string name"),
+                        BuiltInField(definition = "bytes creationCode", enabled = From("0.5.3")),
+                        BuiltInField(definition = "bytes runtimeCode", enabled = From("0.5.3")),
+                        BuiltInField(definition = "bytes4 interfaceId", enabled = From("0.6.7"))
+                    ],
+                    functions = []
+                ),
+                BuiltInType(
+                    name = "%InterfaceTypeType",
+                    fields = [
+                        BuiltInField(definition = "string name"),
+                        BuiltInField(definition = "bytes4 interfaceId", enabled = From("0.6.7"))
+                    ],
+                    functions = []
+                ),
+                BuiltInType(
+                    name = "%IntTypeType",
+                    fields = [
+                        BuiltInField(definition = "int min", enabled = From("0.6.8")),
+                        BuiltInField(definition = "int max", enabled = From("0.6.8"))
+                    ],
+                    functions = []
+                ),
+                BuiltInType(
+                    name = "%UserDefinedValueType",
+                    fields = [],
+                    functions = [
+                        BuiltInFunction(
+                            name = "wrap",
+                            parameters = ["%WrappedType elementaryType"],
+                            return_type = "%UserType"
+                        ),
+                        BuiltInFunction(
+                            name = "unwrap",
+                            parameters = ["%UserType userType"],
+                            return_type = "%WrappedType"
+                        )
+                    ],
+                    enabled = From("0.8.8")
+                ),
+                BuiltInVariable(definition = "%Function _"),
```

Comment:
nit suggestion: adding a comment to explain this:

```suggestion
                // Placeholder for modifiers, to be used to inject the original modified function (`_;`)
                BuiltInVariable(definition = "%Function _"),
```

Reply:
IIUC, `_` is a valid function name in Solidity, and users can invoke it normally.
The below code compiles successfully:

```solidity
contract Foo {
    function _() public {}

    function bar() public {
        _; // refers to the function above
    }
}
```

How do we make sure we bind it only within modifiers? do we have a test for that?